code/game/machinery/autolathe.dm

```diff
@@ -145,88 +145,115 @@
-				var/list/list_to_show = list()
-				//list all materials in said category
-				for(var/i in SSmaterials.materials_by_category[used_material])
-					if(materials.materials[i] > 0)
-						list_to_show += i
-				//ask user to pick specific material from list
-				used_material = tgui_input_list(
-					usr,
-					"Choose [used_material]",
-					"Custom Material",
-					sort_list(list_to_show, GLOBAL_PROC_REF(cmp_typepaths_asc))
-				)
-				if(isnull(used_material))
-					return
-				//the item composition will be made of these materials
-				custom_materials[used_material] += amount_needed
-			materials_used[used_material] = amount_needed
-
-		if(!materials.has_materials(materials_used, coeff, multiplier))
-			say("Not enough materials for this operation!.")
+	var/design_id = params["id"]
+
+	var/valid_design = stored_research.researched_designs[design_id]
+	valid_design ||= stored_research.hacked_designs[design_id]
+	valid_design ||= imported_designs[design_id]
+	if(!valid_design)
+		return
+
+	var/datum/design/design = SSresearch.techweb_design_by_id(design_id)
+	if(isnull(design))
+		stack_trace("got passed an invalid design id: [design_id] and somehow made it past all checks")
+		return
+
+	if(!(design.build_type & AUTOLATHE))
+		return
+
+	var/build_count = text2num(params["multiplier"])
+	if(!build_count)
+		return
+	build_count = clamp(build_count, 1, 50)
+
+	var/list/total_materials_used = list()
+	var/list/materials_per_item = list()
+	var/material_cost_coefficient = ispath(design.build_path, /obj/item/stack) ? 1 : creation_efficiency
+	for(var/datum/material/material as anything in design.materials)
+		var/amount_needed = design.materials[material] * material_cost_coefficient
+		if(istext(material)) // category
+			var/list/choices = list()
+			for(var/datum/material/valid_candidate as anything in SSmaterials.materials_by_category[material])
+				if(materials.get_material_amount(valid_candidate) < amount_needed)
+					continue
+				choices[valid_candidate.name] = valid_candidate
+			if(!length(choices))
+				say("No valid materials with applicable amounts detected for design.")
+				return
+			var/chosen = tgui_input_list(
+				ui.user,
+				"Select the material to use",
+				"Material Selection",
+				sort_list(choices),
+			)
+			if(isnull(chosen))
+				return // user cancelled
+			material = choices[chosen]
+
+		if(isnull(material))
+			stack_trace("got passed an invalid material id: [material]")
 			return
+		total_materials_used[material] = amount_needed * build_count
+		materials_per_item[material] = amount_needed
 
-		//use power
-		var/total_amount = 0
-		for(var/material in being_built.materials)
-			total_amount += being_built.materials[material]
-		use_power(max(active_power_usage, (total_amount) * multiplier / 5))
+	if(!materials.has_materials(total_materials_used))
+		say("Not enough materials to begin production.")
+		return
 
-		//use materials
-		materials.use_materials(materials_used, coeff, multiplier)
```

Comment:
Use the 2nd and 3rd params instead of creating a new list called total_materials and doing all the scaling yourself.

It's just codebloat for no reason.